engine/class_modules/warlock/sc_warlock.hpp

```diff
@@ -974,6 +977,26 @@ struct grimoire_of_sacrifice_damage_t : public warlock_spell_t
   }
 };
 
+struct demonic_synergy_proc_t : public warlock_spell_t
+{
+  demonic_synergy_proc_t( warlock_t* p )
+    : warlock_spell_t( "demonic_synergy_proc", p, p->find_spell( 337060 ) )
+  {
+    background = true;
+    proc = true;
+  }
+
+  void execute() override
+  {
+    if ( p()->warlock_pet_list.active )
```

Comment:
Leaving out the call to parent's execute is generally only safe with ACTION_OTHER (and similar) action types and even there, you need to be careful with it. I'd definitely advise against doing that in subclass of spell_t.